python/src/equistore/tensor.py

```diff
@@ -371,9 +330,54 @@ def components_to_properties(self, variables: Union[str, List[str]]):
         """
         c_variables = _list_or_str_to_array_c_char(variables)
 
-        self._lib.eqs_tensormap_components_to_properties(
+        ptr = self._lib.eqs_tensormap_components_to_properties(
             self._ptr, c_variables, c_variables._length_
         )
+        return TensorMap._from_ptr(ptr)
+
+    def keys_to_properties(
+        self,
+        keys_to_move: Union[str, List[str], Labels],
+        *,
+        sort_samples=True,
+    ) -> "TensorMap":
```

Comment:
Why is this ``"TensorMap"`` and not just ``TensorMap``?

Reply:
Also, I was just wondering out of curiosity if there is a good reason why we are going for ``new_tm = tm.keys_to_properties(keys_to_move)`` over ``new_tm = keys_to_properties(tm, keys_to_move)``, i.e. why keep as a class method if it returns a new ``TensorMap`` object?

Reply:
> Why is this "TensorMap" and not just TensorMap?

I remember either mypy or the Python parser chocking if we use the current class as a return type, I'll have to check again. The string syntax has the same meaning.

> why keep as a class method if it returns a new TensorMap object?

because the function still acts on a TensorMap instance. This is similar to `array.contiguous()` or `array.sum()`, which returns new arrays.

Reply:
Ok makes sense :)